graphql/query_test.go

```diff
@@ -149,4 +153,21 @@ func TestMainlineCommits(t *testing.T) {
 	lastCommit = res.Versions[len(res.Versions)-1].Version
 	assert.NotNil(t, lastCommit)
 	require.Equal(t, utility.FromIntPtr(res.NextPageOrderNumber), lastCommit.Order)
+
+	// Should only return mainline commits that match the passed in requester
+	mainlineCommitOptions.Requesters = []string{evergreen.RepotrackerVersionRequester}
+	res, err = config.Resolvers.Query().MainlineCommits(ctx, mainlineCommitOptions, &buildVariantOptions)
+	require.NoError(t, err)
+	assert.NotNil(t, res)
+
+	require.Equal(t, 8, utility.FromIntPtr(res.NextPageOrderNumber))
+	assert.Nil(t, res.PrevPageOrderNumber)
+	require.Equal(t, 3, len(res.Versions))
+
+	for _, v := range res.Versions {
+		if v.Version != nil {
+			assert.NotNil(t, v.Version)
```

Comment:
Can use require.NotNil if you want to just check if it's nil before 170 (and remove the if statement), otherwise this statement seems redundant